### ISSUE TITLE
Strings: Use comments from .strings file as documentation

### ISSUE THREAD

**Issue by mdznr** [SwiftGen/SwiftGen#563]
It doesn’t currently seem possible to use the comment from within the .strings file as the document comment for the `String` constant in the source code. The current template just uses the translation as the documentation comment instead.

Input (.strings):
```
/* The title of the button used to confirm an action in an alert. */
"Alert.Confirmation" = "OK"
```

Actual Output (.swift):
```
/// OK
internal static let confirmation = Strings.tr("Example", "Alert.confirmation")
```

Expected Output (.swift):
```
/// The title of the button used to confirm an action in an alert.
internal static let confirmation = Strings.tr("Example", "Alert.confirmation")
```

Relevant documentation:
https://github.com/SwiftGen/SwiftGen/blob/master/Documentation/SwiftGenKit%20Contexts/Strings.md#output

**Comment by AliSoftware** [comment]
Hi @mdznr

I agree with the usefulness of such a feature.
However, the implementation of it would be very tricky and error prone.

This is because the `.strings` files are in fact just Plist files in the ASCII format, and thus parsing the keys and values of a `.strings` file in Swift is as easy as calling `NSDictionary(contentsOf: "Localizable.strings")` (which is what we currently use in SwiftGen's code to parse strings files). But of course `NSDictionary(contentsOf: "Localizable.strings")` only returns a dictionary of the keys and values but doesn't include the comments you might have in your strings file.

Being able to parse the comments in strings files would require manually creating a dedicated custom parser to read the strings file line by line instead, and implement all the subtleties of that format, including values spanning on multiple lines, string escapes (like quotes being escaped inside bake strings themselves), various style of comments, and other stuff I'm probably forgetting. Besides, this ASCII Plist format dates from the NeXTStep days and has been officially deprecated since, so a proper documentation of that file format to be sure we cover all possible cases is hard to find.

So overall creating or own parser for strings files, instead of using `NSDictionary(contentsOf: "Localizable.strings")`, seems like a big change in implementation and quite error prone to me, despite the usefulness such a feature would be.

That being said, if you know of a Swift library supporting ASCII Plists with comments that we could integrate, I'd welcome a PR, as this would make a great addition.

**Comment by djbe** [comment]
A PR would definitely be welcome for such a feature, we could then add a `comment` property to each table entry.

It would need to be thoroughly tested though. And do we define as a comment for a particular key? Everything that precedes it, until the previous key/value?

**Comment by CraigSiemens** [comment]
I'd like to have this as well and might have some time to look into it.

Any recommendations for how to go about parsing the comments and keys from the strings file?

I was thinking that using a `Scanner` could walk through the file and parse out the comments and keys. I could see there being some edge cases around keys containing `\"` or other escaped characters. It looks like the `PropertyListDecoder` is doing some work to convert reading the two characters `\"` from the strings file into a single `"` in the in the returned dictionary.

**Comment by ninhgeocomply** [comment]
@CraigSiemens @mdznr 

Xcode 14 has a new feature called "Export Localizations", the exported files can show Comments from .strings files.
Do you think apple will provide some public APIs for this?

Link: https://developer.apple.com/videos/play/wwdc2022/110427/?time=936

<img width="1095" alt="Screen Shot 2022-06-14 at 14 36 01" src="https://user-images.githubusercontent.com/93517594/173521347-d3cfde5e-6013-45ad-8ce5-aa61f0b1a362.png">

<img width="256" alt="Screen Shot 2022-06-14 at 14 36 30" src="https://user-images.githubusercontent.com/93517594/173521370-632c4104-ca83-4570-b0da-bcd236df53e7.png">

**Comment by djbe** [comment]
Implemented by SwiftGen 6.6.0, which has been released.
